src/apps/clients/metadata.clj

```diff
@@ -27,10 +27,11 @@
 (defn get-active-hierarchy-version
   [& {:keys [validate] :or {validate true}}]
   (let [version (db-categories/get-active-hierarchy-version)]
-    (when (and validate (empty? version))
-      (throw+ {:type  :clojure-commons.exception/not-found
-               :error "An app hierarchy version has not been set."}))
-    version))
+    (if (empty? version)
+      (when validate
+        (throw+ {:type  :clojure-commons.exception/not-found
+                 :error "An app hierarchy version has not been set."}))
+      version)))
```

Comment:
This change ensures that `nil` will be returned whenever the active app hierarchy version has not ben set, which makes it easier for `filter-targets-by-ontology-search` to avoid querying the metadata service if there's no active ontology.